src/plopp/backends/plotly/line.py

```diff
@@ -0,0 +1,230 @@
+                                name=self.label,
+                                mode=mode,
+                                marker=marker_style,
+                                line_shape=line_shape,
+                                error_y=error_y,
+                                line=line_style)
+
+        marker_line_style = {'width': 3, 'color': mask_color}
+        if 'line' in marker_style:
+            marker_style['line'].update(marker_line_style)
+        else:
+            marker_style['line'] = marker_line_style
+        if 'width' in line_style:
+            line_style['width'] *= 5
+        else:
+            line_style['width'] = 5
+        if data["hist"]:
+            line_style['color'] = mask_color
+
+        self._mask = go.Scatter(x=np.asarray(data["values"]["x"]),
+                                y=np.asarray(data[mask_data_key]["y"]),
+                                name=self.label,
+                                mode=mode,
+                                marker=marker_style,
+                                line_shape=line_shape,
+                                line=line_style,
+                                visible=has_mask,
+                                showlegend=False)
+
+        # Below, we need to re-define the line because it seems that the Scatter trace
+        # that ends up in the figure is a copy of the one above.
+        # Plotly has no concept of zorder, so we need to add the traces in a specific
+        # order
+        if data["hist"]:
+            self._fig.add_trace(self._mask)
+            self._mask = self._fig.data[-1]
+            self._fig.add_trace(self._line)
+            self._line = self._fig.data[-1]
+        else:
+            self._fig.add_trace(self._line)
+            self._line = self._fig.data[-1]
+            self._fig.add_trace(self._mask)
+            self._mask = self._fig.data[-1]
+
+    def _preprocess_hist(self, data: dict) -> dict:
+        """
+        Convert 1d data to be plotted to internal format, e.g., padding
+        histograms and duplicating info for variances.
+        """
+        x = data["values"]["x"]
+        y = data["values"]["y"]
+        hist = len(x) != len(y)
+        if hist:
+            data["values"]["y"] = np.concatenate((y[0:1], y))
+            if data["mask"] is not None:
+                data["mask"]["y"] = np.concatenate(
+                    (data["mask"]["y"][0:1], data["mask"]["y"]))
+        if self._data.variances is not None:
+            data["variances"]["x"] = 0.5 * (x[1:] + x[:-1]) if hist else x
+        data["variances"]["y"] = y
+        data["hist"] = hist
+        return data
+
+    def _make_data(self) -> dict:
+        data = {"values": {}, "variances": {}, "mask": None}
+        data["values"]["x"] = self._data.meta[self._dim].values
+        data["values"]["y"] = self._data.values
+        if self._data.variances is not None:
+            data["variances"]["e"] = sc.stddevs(self._data.data).values
```

Comment:
Very confusing to call this "variances" then?

Reply:
Why not `data['stddevs']` or `data['errors']`? I think `variances` is very misleading.

Reply:
I don't really see why it is much more misleading than `stddevs` or `errors`. `data` has a `variances` entry if `self._data.variances` is not `None`.